From 2ddcd6b047bb8786e808c4af6c32b6a0dadbcacd Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 23:11:06 -0400
Subject: [ethernet] cxgb4vf: Remove superfluous call to pci_disable_msix()

Message-id: <1410477072-95160-2-git-send-email-svemuri@redhat.com>
Patchwork-id: 93896
O-Subject: [RHEL7.1 PATCH BZ 1070945 1/7] cxgb4vf: Remove superfluous call to pci_disable_msix()
Bugzilla: 1070945
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

There is no need to call pci_disable_msix() in case
the previous call to pci_enable_msix() failed

Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Cc: Casey Leedom <leedom@chelsio.com>
Cc: netdev@vger.kernel.org
Cc: linux-pci@vger.kernel.org
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit f6f273bfe1e9d678de3598feb7906138cfe68b71)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4vf/cxgb4vf_main.c b/drivers/net/ethernet/chelsio/cxgb4vf/cxgb4vf_main.c
index 0899c09..509fed9 100644
--- a/drivers/net/ethernet/chelsio/cxgb4vf/cxgb4vf_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4vf/cxgb4vf_main.c
@@ -2475,7 +2475,6 @@ static int enable_msix(struct adapter *adapter)
   for (i = 0; i < want; ++i)
    adapter->msix_info[i].vec = entries[i].vector;
  } else if (err > 0) {
-  pci_disable_msix(adapter->pdev);
   dev_info(adapter->pdev_dev, "only %d MSI-X vectors left,"
     " not using MSI-X\n", err);
  }
-- 
1.7.1